tests/v2_validation/cattlevalidationtest/core/test_ldap_auth.py

```diff
@@ -1590,3 +1592,126 @@ def test_secret_setting(admin_client):
                               password=ldap_main_pass)
     secret = client.by_id_setting('api.auth.ldap.service.account.password')
     assert secret.value is None
+
+
+# 2
+@if_test_ad
+def test_ad_restrict_to_specific_user(admin_client):
+    ldap_main_user = os.environ.get('AD_MAIN_USER')
+    ldap_main_pass = os.environ.get('AD_MAIN_PASS')
+    ldap_user2 = os.environ.get('AD_USER2')
+    ldap_pass2 = os.environ.get('AD_PASS2')
+    ldap_user3 = os.environ.get('AD_USER3')
+    ldap_pass3 = os.environ.get('AD_PASS3')
+    ldap_port = os.environ.get('LDAP_PORT')
+
+    user = ADMIN_TOKEN['userIdentity']
+    allowed_identities = []
+    allowed_identities.append(user)
+    token2 = get_authed_token(username=ldap_user2,
+                              password=ldap_pass2)
+    token3 = get_authed_token(username=ldap_user3,
+                              password=ldap_pass3)
+    user2 = token2['userIdentity']
+    allowed_identities.append(user2)
+
+    # Enable new configuration
+    config = load_config(access_mode='required')
+    config['enabled'] = True
+    config['allowedIdentities'] = allowed_identities
+
+    if ldap_port == 'True':
+        test_config = load_test_api_config(config)
+        access_key = ADMIN_AD_CLIENT._access_key
+        secret_key = ADMIN_AD_CLIENT._secret_key
+        auth_url = cattle_url()[:-7] + 'v1-auth/testlogin'
+        r = requests.post(auth_url, data=json.dumps(test_config),
+                          auth=(access_key, secret_key))
+        assert r.ok
+        auth_url = cattle_url()[:-7] + 'v1-auth/config'
+        r = requests.post(auth_url, data=json.dumps(config),
+                          auth=(access_key, secret_key))
+        assert r.ok
+    else:
+        ADMIN_AD_CLIENT.create_ldapconfig(config)
+
+    cookies = dict(token=token3['jwt'])
+    bad_auth = requests.get(cattle_url() + "schemas", cookies=cookies)
+    assert bad_auth.status_code == 403
+
+    cookies = dict(token=token2['jwt'])
+    good_auth = requests.get(cattle_url() + "schemas", cookies=cookies)
+    assert good_auth.status_code == 200
+
+    config = load_config()
+    config['enabled'] = None
+    if ldap_port == 'True':
```

Comment:
i guess there is no need to disable it, however i think it should be returned back to unrestricted, because the finalizer is called after all the tests are passed.